backend/tests/clubs/test_serializers.py

```diff
@@ -0,0 +1,152 @@
+
+    def test_member_sees_current_version(self):
+        Membership.objects.create(person=self.user, club=self.club)
+        request = self.factory.get(reverse("clubs-detail", args=[self.club.code]))
+        request.user = self.user
+
+        self.serializer.context["request"] = Request(request)
+        result = self.serializer.to_representation(self.club)
+        self.assertEqual(result["name"], "Test Club")
+
+    def test_approved_club_visible_to_all(self):
+        request = self.factory.get(reverse("clubs-detail", args=[self.club.code]))
+        request.user = self.user
+
+        self.serializer.context["request"] = Request(request)
+        result = self.serializer.to_representation(self.club)
+        self.assertEqual(result["name"], "Test Club")
+
+    def test_bypass_flag_shows_current_version(self):
+        request = self.factory.get(
+            f"{reverse('clubs-detail', args=[self.club.code])}?bypass=true"
+        )
+        request.user = self.user
+
+        self.serializer.context["request"] = Request(request)
+        result = self.serializer.to_representation(self.club)
+        self.assertEqual(result["name"], "Test Club")
+
+    def test_non_privileged_user_during_renewal_period(self):
+        # Simulate deactivation at the beginning of the renewal period
+        self.club.save()
+        self.club.approved = False
+        self.club.save()
+
+        # Insert a previously approved version into the club's history
+        past_time = timezone.now() - timedelta(days=180)
+        with mock.patch("django.utils.timezone.now", return_value=past_time):
+            self.club.name = "Past Approved Version"
+            self.club.approved = True
+            self.club.approved_on = past_time
+            self.club.save()
+
+        self.club.name = "Current Unapproved Version"
+        self.club.approved = False
+        self.club.approved_on = None
+        self.club.save()
+
+        request = self.factory.get("/fake-url/")
+        request.user = self.user
+        serializer = ClubSerializer(context={"request": Request(request)})
+
+        result = serializer.to_representation(self.club)
+        self.assertEqual(result["name"], "Past Approved Version")
+        self.assertNotEqual(result["name"], "Current Unapproved Version")
+
+    def test_non_privileged_user_outside_renewal_period_ghost_club(self):
+        self.club.approved = False
+        self.club.ghost = True
+        self.club.save()
+
+        current_date = self.renewal_end + timedelta(days=1)
+        with mock.patch("django.utils.timezone.now", return_value=current_date):
+            approved_date = self.renewal_start + timedelta(days=1)
+            with mock.patch("django.utils.timezone.now", return_value=approved_date):
+                self.club.name = "Current Cycle Approved Version"
+                self.club.approved = True
+                self.club.approved_on = approved_date
+                self.club.save()
```

Comment:
Unless there is internal logic in `club.save` or `serializer.to_representation` that uses `django.utils.timezone.now`, we might not need the `with` clauses since we are explicitly setting `club.approved_on`